source/fundamentals/builders/aggregation-expression-operations.txt

```diff
@@ -0,0 +1,1198 @@
+     - *No corresponding operator*
+
+.. include:: /includes/fundamentals/agg-mql-diff.rst
+
+Refer to any of the other sections for examples using these convenience
+operators.
+
+Conversion Operations
+~~~~~~~~~~~~~~~~~~~~~
+
+You can perform a conversion operation to convert between ``MqlValue``
+types using the methods described in this section.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 50 50
+
+   * - Java Method
+     - Aggregation Pipeline Operator
+
+   * - `asDocument() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlMap.html#asDocument()>`__
+     - *No corresponding operator*
+       
+   * - `asMap() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#asMap()>`__
+     - *No corresponding operator*
+
+   * - `asString() for MqlDate <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDate.html#asString(com.mongodb.client.model.mql.MqlString,com.mongodb.client.model.mql.MqlString)>`__
+     - :manual:`$dateToString </reference/operator/aggregation/dateToString/>`
+
+   * - `asString() for MqlValue <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#asString()>`__
+     - :manual:`$toString </reference/operator/aggregation/toString/>`
+
+   * - `millisecondsAsDate() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlInteger.html#millisecondsAsDate()>`__
+     - :manual:`$toDate </reference/operator/aggregation/toDate/>`
+
+   * - `parseDate() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#parseDate()>`__
+     - :manual:`$dateFromString </reference/operator/aggregation/dateFromString/>`
+
+   * - `parseInteger() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#parseInteger()>`__
+     - :manual:`$toInt </reference/operator/aggregation/toInt/>`
+
+.. include:: /includes/fundamentals/agg-mql-diff.rst
+
+Suppose you want to have a collection of student data that includes
+their graduation years, which are stored as strings. You want to
+calculate the year of their five-year reunion and store this value in a
+new field.
+
+The ``parseInteger()`` method converts the ``graduationYear`` to an integer
+so that ``add()`` can calculate the reunion year. The ``addFields()`` method
+stores this result as a new ``reunionYear`` field.
+
+The following code shows the pipeline for this aggregation:
+
+.. code-block:: java
+   :copyable: true
+
+   asList(addFields(
+         new Field("reunionYear", current()
+               .getString("graduationYear")
+               .parseInteger()
+               .add(5))
+   ));
+
+The following code provides an equivalent aggregation pipeline in
+the Query API:
+
+.. code-block:: javascript
+   :copyable: true
+
+   [
+     {
+       $addFields: {
+         reunionYear: {
+           $add: [ { $toInt: "$graduationYear" }, 5 ]}}
+   }]
+
+Custom Operations
+~~~~~~~~~~~~~~~~~
+
+You can apply custom functions to values of type
```

Comment:
I would prefer to keep the custom operations section separate, perhaps we can rename "Convenience" to something else. Or, we could move this to a separate section altogether and call it Representation of Literals or something like that.

Reply:
I agree about separate - they do look different. It just seems that "pass..." is definitely a "convenience" method (operation?)

But the things being called convenience at the moment are... I don't know what to call them. "Conversions"? But we have a section like that. Values? I guess they are not literals from the perspective of Java (since one can put `of(myVariable)`).